plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java

```diff
@@ -360,6 +361,16 @@ public boolean isDefaultHost() {
 		return false;
 	}
 
+	@Override
+	public String getConsoleURL(IProgressMonitor monitor) {
+		return client.adapt(IConsole.class).getConsoleUrl();
+	}
+
+	@Override
+	public <R extends IResource> String getConsoleURL(R resource, IProgressMonitor monitor) {
```

Comment:
What's the added value of using generic type instead of IResource ?

Reply:
with generics you get the exact type, what's the drawback?

Reply:
I don't see the added value as having the parameter typed as IResource. Generics are usefull in the case you want to return the same type for example but here you are returning a String so no need